src/coordinator/api/v1/handler/database/create.go

```diff
@@ -246,42 +294,50 @@ func defaultedPlacementInitRequest(r *admin.DatabaseCreateRequest, dbCfg dbconfi
 				IsolationGroup: "local",
 				Zone:           "embedded",
 				Weight:         1,
-				Endpoint:       fmt.Sprintf("http://localhost:%d", port),
+				Endpoint:       fmt.Sprintf("127.0.0.1:%d", port),
 				Hostname:       "localhost",
 				Port:           uint32(port),
 			},
 		}
 	case dbTypeCluster:
-		// This function assumes only one of Hostnames/HostnameGroups have been set
-		numHosts := len(r.Hostnames) + len(r.HostnameGroups)
+
+		numHosts := len(r.Hosts)
 		numShards = int32(math.Min(math.MaxInt32, powerOfTwoAtLeast(float64(numHosts*shardMultiplier))))
 		replicationFactor = r.ReplicationFactor
 		if replicationFactor == 0 {
 			replicationFactor = 3
 		}
+
 		instances = make([]*placementpb.Instance, 0, numHosts)
 
-		for _, hostname := range r.Hostnames {
-			instances = append(instances, &placementpb.Instance{
-				Id:             hostname,
-				IsolationGroup: "cluster",
-				Zone:           "embedded",
-				Weight:         1,
-				Endpoint:       fmt.Sprintf("http://%s:%d", hostname, port),
-				Hostname:       hostname,
-				Port:           uint32(port),
-			})
-		}
+		for _, host := range r.Hosts {
+			id := strings.TrimSpace(host.Id)
+			if id == "" {
+				return nil, errMissingHostID
+			}
+
+			isolationGroup := strings.TrimSpace(host.IsolationGroup)
+			if isolationGroup == "" {
+				isolationGroup = defaultIsolationGroup
+			}
+
+			zone := strings.TrimSpace(host.Zone)
+			if zone == "" {
+				zone = defaultZone
+			}
+
+			weight := host.Weight
+			if weight == 0 {
+				weight = 1
+			}
 
-		for _, hostnameGroup := range r.HostnameGroups {
 			instances = append(instances, &placementpb.Instance{
-				Id:             hostnameGroup.Hostname,
-				IsolationGroup: hostnameGroup.IsolationGroup,
-				Zone:           "embedded",
-				Weight:         1,
-				Endpoint:       fmt.Sprintf("http://%s:%d", hostnameGroup.Hostname, port),
-				Hostname:       hostnameGroup.Hostname,
-				Port:           uint32(port),
+				Id:             id,
+				IsolationGroup: isolationGroup,
+				Zone:           zone,
+				Weight:         weight,
+				Endpoint:       fmt.Sprintf("%s:%d", host.Address, host.Port),
```

Comment:
It can be a hostname or an IP address unfortunately.  So in that case, it's tough to know whether its a hostname (alphanumeric) or IPv4 or IPv6.  To do validation may cause valid hostnames to be deemed invalid potentially I believe?